src/platform_impl/linux/x11/util/cursor.rs

```diff
@@ -1,12 +1,63 @@
-use std::ffi::CString;
-use std::iter;
+use std::{ffi::CString, iter, slice};
 
 use x11rb::connection::Connection;
 
-use crate::window::CursorIcon;
+use crate::{cursor::CursorImage, window::CursorIcon};
 
 use super::*;
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct CustomCursorInternal(ffi::Cursor);
```

Comment:
At the moment we're working on phasing out Xlib (#3198). So I would prefer using `x11rb`'s utilities instead. It shouldn't be too complicated:

- Use [`create_pixmap`](https://docs.rs/x11rb/latest/x11rb/protocol/xproto/fn.create_pixmap.html) then [`put_image`](https://docs.rs/x11rb/latest/x11rb/protocol/xproto/fn.put_image.html) to create a pixmap with the cursor's contents.
- Use [`create_cursor`](https://docs.rs/x11rb/latest/x11rb/protocol/xproto/fn.create_cursor.html) to turn that pixmap into a cursor.
- [Change the window attributes](https://docs.rs/x11rb/latest/x11rb/protocol/xproto/struct.ChangeWindowAttributesAux.html#method.cursor) to use this cursor.

Reply:
I tried to do it with create_cursor, but it seems to actually only be able to show cursors with two colors instead of a full rgba image https://www.x.org/releases/X11R7.7/doc/xproto/x11protocol.html#requests:CreateCursor. And I couldn't find an equivalent to the XcursorImageCreate in x11rb.

Reply:
It probably requires some subtlety; this is how `XcursorImageCreate` handles it under the hood, after all. I've been meaning to remove the `Xcursor` bits in a future PR. So this is fine for now.